xonsh/completers/_aliases.py

```diff
@@ -0,0 +1,125 @@
+import builtins
+
+from collections import OrderedDict
+
+from xonsh.completers.tools import justify
+
+
+def _add_one_completer(name, func, loc='end'):
+    new = OrderedDict()
+    if loc == 'start':
+        new[name] = func
+        for (k, v) in builtins.__xonsh_completers__.items():
+            new[k] = v
+    elif loc == 'end':
+        for (k, v) in builtins.__xonsh_completers__.items():
+            new[k] = v
+        new[name] = func
+    else:
+        dir, rel = loc[0], loc[1:]
+        found = False
+        for (k, v) in builtins.__xonsh_completers__.items():
+            if rel == k and dir == '<':
+                new[name] = func
+                found = True
+            new[k] = v
+            if rel == k and dir == '>':
+                new[name] = func
+                found = True
+        if not found:
+            new[name] = func
+    builtins.__xonsh_completers__ = new
```

Comment:
Maybe a clear() and an update() would be better than complete replacement.  That way, folks that have a reference to the original object would not loose it.  Namely,

```python
builtins.__xonsh_completers__.clear()
builtins.__xonsh_completers__.update(new)
```

You could also change new to just be a list of tuples.